### ISSUE TITLE
Create mapeo-core/discovery.js module as wrapper around mdns-sd-discovery and hyperswarm

### ISSUE THREAD

**Issue by sethvincent** [digidem/mapeo-core-next#6]
## Description
Implement a way to find & sync with peers via both mdns and hyperswarm

Dependencies:
- https://github.com/digidem/mdns-sd-discovery/
- https://github.com/hyperswarm/hyperswarm/

## Tasks
- [ ] Create `Discovery` class and methods
- [ ] Write tests

**Comment by sethvincent** [comment]
An update on this:

Previously I had been working toward syncing one project at a time (the active project) where discovery code is part of a `Project` class:

```js
import Mapeo from '@mapeo/core'

const rootKey = Mapeo.getRootKey() 
const mapeo = new Mapeo({ rootKey })
const project = mapeo.project(name)

await project.ready()
const progress = project.sync()

progress.on('status', (status) => {})

project.on('peer', (peer) => {})
```

But as we talk about this and I get into implementation, it makes more sense to be able to sync multiple projects (or just one project) where discovery and syncing is part of the mapeo class rather than the project class with something like this:

```js
import Mapeo from '@mapeo/core'

const rootKey = Mapeo.getRootKey() 
const mapeo = new Mapeo({ rootKey })
const project1 = mapeo.project(name1)
const project2 = mapeo.project(name2)

await project1.ready()
await project2.ready()

const progress = mapeo.sync([
	project1,
	project2
])

progress.on('status', (status) => {
  // `status` is something like an array of percentages for each project
  // or an array of objects with sync progress for each peer in each project
})

mapeo.on('peer', (peer) => {
  // each peer would have information on what projects they are part of and their sync state
  // same info could be retrieved via a getter like `mapeo.peers`
})
```

These are just rough sketches!

**Comment by sethvincent** [comment]
closed by #11
